src/map/map_state.cpp

```diff
@@ -98,6 +98,7 @@ void map_state::update(sys::state& state) {
 		last_update_time = now;
 
 	update_unit_arrows(state, map_data);
+	map_data.update_fog_of_war(state);
```

Comment:
Doing this here results in the fog being recalculated every frame. It only needs to be recalculated when the game state changes